interfaces/builtin/netlink_driver_test.go

```diff
@@ -0,0 +1,262 @@
+	s.gadgetBadNumberSlot = interfaces.NewConnectedSlot(s.gadgetBadNumberSlotInfo, nil, nil)
+	s.gadgetBadNameSlotInfo = gadgetInfo.Slots["bad-family-name"]
+	s.gadgetBadNameSlot = interfaces.NewConnectedSlot(s.gadgetBadNameSlotInfo, nil, nil)
+	s.gadgetBadNameStringSlotInfo = gadgetInfo.Slots["bad-family-name-string"]
+	s.gadgetBadNameStringSlot = interfaces.NewConnectedSlot(s.gadgetBadNameStringSlotInfo, nil, nil)
+
+	osInfo := snaptest.MockInfo(c, `
+name: my-core
+version: 0
+type: os
+slots:
+    my-driver:
+        interface: netlink-driver
+        family: 777
+        family-name: seven-7-seven
+`, nil)
+	s.osNetlinkSlotInfo = osInfo.Slots["my-driver"]
+	s.osNetlinkSlot = interfaces.NewConnectedSlot(s.osNetlinkSlotInfo, nil, nil)
+
+	// Snap Consumers
+	consumingSnapInfo := snaptest.MockInfo(c, `
+name: client-snap
+version: 0
+plugs:
+  plug-for-netlink-driver-777:
+    interface: netlink-driver
+    family-name: seven-7-seven
+  plug-for-netlink-driver-foo:
+    interface: netlink-driver
+    family-name: foo-driver
+  missing-family-name:
+    interface: netlink-driver
+  invalid-family-name:
+    interface: netlink-driver
+    family-name: ---foo-----
+  invalid-family-name-string:
+    interface: netlink-driver
+    family-name: 1213123
+apps:
+  netlink-test:
+    command: bin/foo.sh
+    plugs: 
+      - plug-for-netlink-driver-777
+      - plug-for-netlink-driver-foo
+`, nil)
+	s.appToCorePlugDriverInfo = consumingSnapInfo.Plugs["plug-for-netlink-driver-777"]
+	s.appToCorePlugDriver = interfaces.NewConnectedPlug(s.appToCorePlugDriverInfo, nil, nil)
+
+	s.appToGadgetPlugDriverInfo = consumingSnapInfo.Plugs["plug-for-netlink-driver-foo"]
+	s.appToGadgetPlugDriver = interfaces.NewConnectedPlug(s.appToCorePlugDriverInfo, nil, nil)
+
+	s.appBadFamilyNamePlugDriverInfo = consumingSnapInfo.Plugs["invalid-family-name"]
+	s.appBadFamilyNamePlugDriver = interfaces.NewConnectedPlug(s.appBadFamilyNamePlugDriverInfo, nil, nil)
+
+	s.appBadFamilyNameStringPlugDriverInfo = consumingSnapInfo.Plugs["invalid-family-name-string"]
+	s.appBadFamilyNameStringPlugDriver = interfaces.NewConnectedPlug(s.appBadFamilyNameStringPlugDriverInfo, nil, nil)
+
+	s.appMissingFamilyNamePlugDriverInfo = consumingSnapInfo.Plugs["missing-family-name"]
+	s.appMissingFamilyNamePlugDriver = interfaces.NewConnectedPlug(s.appMissingFamilyNamePlugDriverInfo, nil, nil)
+}
+
+func (s *NetlinkDriverInterfaceSuite) TestStaticInfo(c *C) {
+	si := interfaces.StaticInfoOf(s.iface)
+	c.Assert(si.ImplicitOnCore, Equals, false)
+	c.Assert(si.ImplicitOnClassic, Equals, false)
+	c.Assert(si.Summary, Equals, "allows operating a kernel driver module exposing itself via a netlink protocol family")
+	c.Assert(si.BaseDeclarationSlots, testutil.Contains, "netlink-driver")
+}
+
+func (s *NetlinkDriverInterfaceSuite) TestAutoConnect(c *C) {
+	// with matching family-name attributes, it works
+	c.Check(s.iface.AutoConnect(s.appToCorePlugDriverInfo, s.osNetlinkSlotInfo), Equals, true)
```

Comment:
(nitpick^3) I was wondering if we should do something like:
```
func (s *NetlinkDriverInterfaceSuite) TestAutoConnect(c *C) {
        // with matching family-name attributes, it works
+       c.Check(s.appToCorePlugDriverInfo.Attrs["family-name"], Equals, "seven-7-seven")
+       c.Check(s.osNetlinkSlotInfo.Attrs["family-name"], Equals, "seven-7-seven")
```
to gently remind the reader of the code what the family-name is that is matched here in auto-connect. But certainly should not hold back this PR :)

Reply:
sure

Reply:
done in https://github.com/snapcore/snapd/pull/10406